frontend/src/features/public-form/components/FormStartPage/FormHeader.tsx

```diff
@@ -80,13 +91,54 @@ export const MiniHeader = ({ isOpen }: MiniHeaderProps): JSX.Element | null => {
       ref={miniHeaderRef}
       direction="top"
       in={isOpen}
-      style={{ zIndex: 10 }}
+      style={{ zIndex: 1000 }}
     >
-      <Box bg={titleBg} px="2rem" py="1rem">
+      <Box
+        bg={titleBg}
+        px={{ base: '1.5rem', md: '2rem' }}
+        py={{ base: '0.5rem', md: '1rem' }}
+      >
         <Skeleton isLoaded={!!title}>
-          <Text as="h2" textStyle="h2" textAlign="start" color={titleColour}>
-            {title ?? 'Loading title'}
-          </Text>
+          <Flex
+            align="center"
+            flex={1}
+            gap="0.5rem"
+            justify="space-between"
+            flexDir="row"
+          >
+            <Flex alignItems="center" minH={{ base: '4rem', md: '0' }}>
+              <Text
+                textStyle={{ base: 'h4', md: 'h2' }}
+                textAlign="start"
+                color={titleColour}
+              >
+                {title ?? 'Loading title'}
+              </Text>
+            </Flex>
+            <IconButton
+              variant="solid"
+              colorScheme="primary"
+              aria-label="Mobile section sidebar"
+              fontSize="1.5rem"
+              icon={<BxMenuAltLeft />}
+              d={activeSectionId ? { base: 'flex', md: 'none' } : 'none'}
```

Comment:
question: any reason why we did not use conditional rendering instead? (the commented out code) 